src/chains/filecoin/filecoin/src/data-managers/private-key-manager.ts

```diff
@@ -0,0 +1,59 @@
+import { LevelUp } from "levelup";
+
+const NOTFOUND = 404;
+
+export default class PrivateKeyManager {
+  static AccountsWithPrivateKeysKey = Buffer.from("accounts-with-private-keys");
+  private base: LevelUp;
+
+  static async initialize(base: LevelUp) {
+    const manager = new PrivateKeyManager(base);
+    return manager;
+  }
+
+  constructor(base: LevelUp) {
+    this.base = base;
+  }
+
+  async getPrivateKey(address: string): Promise<string | null> {
+    try {
+      const privateKey: Buffer = await this.base.get(Buffer.from(address));
+      return privateKey.toString();
+    } catch (e) {
+      if (e.status === NOTFOUND) {
+        return null;
+      }
+      throw e;
+    }
+  }
+
+  async putPrivateKey(address: string, privateKey: string) {
+    await this.base.put(Buffer.from(address), Buffer.from(privateKey));
+    const addresses = await this.getAddressesWithPrivateKeys();
+    if (!addresses.includes(address)) {
+      addresses.push(address);
+      await this.base.put(
+        PrivateKeyManager.AccountsWithPrivateKeysKey,
+        Buffer.from(JSON.stringify(addresses))
+      );
```

Comment:
I seeeee yoouuuuu... but choose to look away because it's highly unlikely for the number of addresses to grow large enough to cause problems.

Maybe a little `// TODO(perf)` comment and a new Issue labeled with a low "priority" would be enough for now?